e2e_playwright/st_file_uploader_test.py

```diff
@@ -265,7 +265,60 @@ def test_does_not_call_callback_when_not_changed(app: Page):
     )
 
 
+def test_works_inside_form(app: Page):
+    """Test that uploading a file inside form works as expected."""
+    file_name1 = "form_file1.txt"
+    file_content1 = b"form_file1content"
+
+    uploader_index = 3
+
+    with app.expect_file_chooser() as fc_info:
+        app.get_by_test_id("stFileUploadDropzone").nth(uploader_index).click()
+
+    file_chooser = fc_info.value
+    file_chooser.set_files(
+        files=[{"name": file_name1, "mimeType": "text/plain", "buffer": file_content1}]
+    )
+    wait_for_app_run(app)
+    app.wait_for_timeout(1000)
```

Comment:
nit: these can also be moved to the `wait_delay` parameter of `wait_for_app_run`